### PR TITLE
[IMP] gmail, outlook: hide the create button if we do not have access right

### PR DESCRIPTION
**Pull request by std-odoo**
Purpose
=======
Hide the "Create Partner" button if we do not have the "create" access
right.

Do not offer to create a project if the current user can't.

Task-2826471

### CODE REVIEW

File: gmail/src/views/create_task.ts [hunk 1]
```diff
@@ -130,6 +133,20 @@ export function buildCreateTaskView(state: State, query: string = "", hideCreate
...
+    } else if (noProject) {
+        const noProjectSection = CardService.newCardSection();
+
+        noProjectSection.addWidget(CardService.newImage().setImageUrl(UI_ICONS.empty_folder));
+
+        noProjectSection.addWidget(CardService.newTextParagraph().setText("<b>" + _t("No project") + "</b>"));
+
+        noProjectSection.addWidget(
+            CardService.newTextParagraph().setText(
+                _t("There are no Projects in your Database. Please ask your project manager to create one."),
```
Comment (tde-banana-odoo) [inline gmail/src/views/create_task.ts:145]: Why so much capital letters ^^ ?

File: gmail/src/views/leads.ts [hunk 2]
```diff
@@ -105,8 +108,12 @@ export function buildLeadsView(state: State, card: Card) {
                 ),
             );
         }
-    } else {
+    } else if (state.canCreatePartner) {
         leadsSection.addWidget(CardService.newTextParagraph().setText(_t("Save Contact to create new Opportunities.")));
+    } else {
+        leadsSection.addWidget(
+            CardService.newTextParagraph().setText(_t("The Contact needs to exist to create Opportunities.")),
```
Comment (tde-banana-odoo) [inline gmail/src/views/leads.ts:115]: It is a bit strangely said ... Maybe something like "You can only create opportunities for existing customers" ?

File: gmail/src/views/tasks.ts [hunk 3]
```diff
@@ -74,8 +77,12 @@ export function buildTasksView(state: State, card: Card) {
                 ),
             );
         }
-    } else {
+    } else if (state.canCreatePartner) {
         tasksSection.addWidget(CardService.newTextParagraph().setText(_t("Save the contact to create new tasks.")));
+    } else {
+        tasksSection.addWidget(
+            CardService.newTextParagraph().setText(_t("The Contact needs to exist to create Tasks.")),
```
Comment (tde-banana-odoo) [inline gmail/src/views/tasks.ts:84]: Would update a bit the message here also, if possible :)

File: gmail/src/views/tickets.ts [hunk 4]
```diff
@@ -87,8 +90,12 @@ export function buildTicketsView(state: State, card: Card) {
                 ),
             );
         }
-    } else {
+    } else if (state.canCreatePartner) {
         ticketsSection.addWidget(CardService.newTextParagraph().setText(_t("Save the contact to create new tickets.")));
+    } else {
+        ticketsSection.addWidget(
+            CardService.newTextParagraph().setText(_t("The Contact needs to exist to create Tickets.")),
```
Comment (tde-banana-odoo) [inline gmail/src/views/tickets.ts:97]: And here ^^

File: gmail/src/main.ts [hunk 5]
```diff
@@ -26,7 +26,16 @@ function onGmailMessageOpen(event) {
         throw new Error(_t("Error during enrichment"));
     }
 
-    const state = new State(partner, currentEmail, odooUserCompanies, null, null, error);
+    const state = new State(
```
Comment (tde-banana-odoo) [inline gmail/src/main.ts:29]: Such parameters ... becoming complicated to remember every parameter meaning ^^

File: outlook/src/taskpane/components/SectionTasks/SectionTasks.tsx [hunk 6]
```diff
@@ -50,13 +52,15 @@ class SectionTasks extends React.Component<SectionTasksProps, SectionTasksState>
...
                     partner={this.props.partner}
+                    canCreatePartner={this.props.canCreatePartner}
                     model="project.task"
                     odooEndpointCreateRecord={api.createTask}
                     odooRecordIdName="task_id"
                     odooRedirectAction="project_mail_plugin.project_task_action_form_edit"
                     title="Tasks"
                     titleCount="Tasks (%(count)s)"
                     msgNoPartner="Save Contact to create new Tasks."
+                    msgNoPartnerNoAccess="The Contact needs to exist to create Tasks."
```
Comment (tde-banana-odoo) [inline outlook/src/taskpane/components/SectionTasks/SectionTasks.tsx:63]: And here :D

File: outlook/src/taskpane/components/SectionTasks/SelectProjectDropdown.tsx [hunk 7]
```diff
@@ -129,6 +126,12 @@ class SelectProjectDropdown extends React.Component<SelectProjectProps, SelectPr
                         {_t('Create %(name)s', { name: this.state.query })}
                     </div>
                 )}
+                {this.state.query.length && !allowCreateNewProject && !this.state.projects.length ? (
+                    <div>{_t('No Projects Found')}</div>
```
Comment (tde-banana-odoo) [inline outlook/src/taskpane/components/SectionTasks/SelectProjectDropdown.tsx:130]: Do we need a "s" ? No project found maybe ?

**Review by tde-banana-odoo — COMMENTED** [review]
Quick check, code looks good to me, few comments :)

File: gmail/src/models/partner.ts [hunk 8]
```diff
@@ -78,15 +85,15 @@ export class Partner {
      * If we are not logged to an Odoo database, enrich the email domain with IAP.
      * Otherwise fetch the partner on the user database.
      */
-    static enrichPartner(email: string, name: string): [Partner, number[], ErrorMessage] {
+    static enrichPartner(email: string, name: string): [Partner, number[], boolean, boolean, ErrorMessage] {
```
[the remark this answers is not in the document]
Comment (std-odoo) [reply gmail/src/models/partner.ts:90]: ok :)

**Review by std-odoo — COMMENTED** [review]
@tde-banana-odoo Changes done :)

**Pull request merged**